components/automate-ui/src/app/entities/rules/rule.reducer.ts

```diff
@@ -69,15 +71,17 @@ export const ruleAttributes: RuleTypeMappedObject = {
 
 export const ruleEntityAdapter: EntityAdapter<Rule> = createEntityAdapter<Rule>();
 
-export const RuleEntityInitialState: RuleEntityState = ruleEntityAdapter.getInitialState({
-  getAttributes: ruleAttributes,
-  getAllStatus: EntityStatus.notLoaded,
-  getStatus: EntityStatus.notLoaded,
-  createStatus: EntityStatus.notLoaded,
-  createError: null,
-  deleteStatus: EntityStatus.notLoaded,
-  updateStatus: EntityStatus.notLoaded
-});
+export const RuleEntityInitialState: RuleEntityState =
+  ruleEntityAdapter.getInitialState(<RuleEntityState>{
```

Comment:
Now here is a TypeScript anomaly, which I unearthed thanks to @srenatus' question. Originally L83 read
```
projectRulesStatus: <ProjectStatus>'PROJECT_RULES_STATUS_UNSET'
```
And he wondered why. Answer: Because it gave me a compiler error without it, because `getInitialState` takes an untyped object. So I added `<RuleEntityState>` above as a more general solution...
But later I checked further. All of our other reducers using `getInitialState` do **not** add the type assertion, but if I were to, for instance, misspell `getAllStatus` it would still flag as an error in VSCode, thus recognizing it was in fact an _xyz_`EntityState`. What's different here? The `'PROJECT_RULES_STATUS_UNSET'` string value that is both (a) a supertype of the allowed `ProjectStatus` type and (b) not null. Not satisfying those seems to be why I need the type assertion.

Reply:
Interesting. And weird. Shall we contact some upstream?